shared/src/main/scala/pl/metastack/metarx.scala

```diff
@@ -12,4 +12,40 @@ package object metarx extends OptImplicits {
       opt := Some(t)
     }
   }
+
+  implicit class ReadChannelBooleanExtensions(rch: ReadChannel[Boolean]) {
+    def withOperation(other: ReadChannel[Boolean], operator: (Boolean, Boolean) => Boolean) = {
+      rch.flatMap(thisVal => other.map(otherVal => operator(thisVal, otherVal)))
+    }
+    def withOperation(argument: Boolean, operator: (Boolean, Boolean) => Boolean): ReadChannel[Boolean] = {
+      rch.map(value => operator(value, argument))
+    }
+
+    private def andFunc = (a: Boolean, b: Boolean) => a && b
+    def &&(other: ReadChannel[Boolean]): ReadChannel[Boolean] =  {
+      withOperation(other, andFunc)
```

Comment:
You are right! I thought it was working before that change. Seems like the compiler is not smart enough to find the unique function match based on the other parameter...